river/Cursor.zig

```diff
@@ -747,26 +878,74 @@ fn processMotion(self: *Self, device: *wlr.InputDevice, time: u32, delta_x: f64,
         },
         .resize => |data| {
             const border_width = if (data.view.draw_borders) server.config.border_width else 0;
-
-            // Set width/height of view, clamp to view size constraints and output dimensions
             const box = &data.view.pending.box;
-            box.width = @intCast(u32, math.max(0, @intCast(i32, box.width) + @floatToInt(i32, delta_x)));
-            box.height = @intCast(u32, math.max(0, @intCast(i32, box.height) + @floatToInt(i32, delta_y)));
+            const constraints = data.view.getConstraints();
+            const min_width = @intCast(i32, math.max(View.min_size, constraints.min_width));
+            const min_height = @intCast(i32, math.max(View.min_size, constraints.min_height));
+
+            // Using two points instead of a box makes this logic a bit more
+            // readable for humans. The compiler should optimize the conversion out.
+            var botright_x = box.x + @intCast(i32, box.width);
+            var botright_y = box.y + @intCast(i32, box.height);
+
+            if (data.resize_direction == .top or data.resize_direction == .topleft or data.resize_direction == .topright) {
+                const min_y = math.max(@intCast(i32, border_width), botright_y - @intCast(i32, @truncate(u16, constraints.max_height)));
```

Comment:
```suggestion
                const min_y = math.max(@intCast(i32, border_width), botright_y - @as(i32, @truncate(u16, constraints.max_height)));
```

Also, we could move the `@intCast()` for border_width up to where it is defined, or perhaps even better just make it a u16 in `Config`.

It might make sense to move the casting for max_height up by the casting for min_height.